### PR TITLE
Fix centrifuge

### LINKED ISSUE #597: Centrifuge error - (ERR): mkfifo(/tmp/46.inpipe1) failed.
### Description of the bug

I tried to run nf-core/mag with the `test` profile on a new compute cluster and got the following error:
```
-[nf-core/mag] Pipeline completed with errors-
WARN: Singularity cache directory has not been defined -- Remote image will be stored in the path: /mnt/lustre/groups/nahnsen/nahkr648/test/work/singularity -- Use the environment variable NXF_SINGULARITY_CACHEDIR to specify a different location
ERROR ~ Error executing process > 'NFCORE_MAG:MAG:CENTRIFUGE (test_minigut-minigut_cf)'

Caused by:
  Process `NFCORE_MAG:MAG:CENTRIFUGE (test_minigut-minigut_cf)` terminated with an error exit status (17)

Command executed:

  centrifuge -x "minigut_cf"         -p 8         --report-file report.txt         -S results.txt         -1 "test_minigut_1.merged.fastq.gz" -2 "test_minigut_2.merged.fastq.gz"
  centrifuge-kreport -x "minigut_cf" results.txt > test_minigut.centrifuge_kreport.txt
  cat results.txt | cut -f 1,3 > results.krona

  cat <<-END_VERSIONS > versions.yml
  "NFCORE_MAG:MAG:CENTRIFUGE":
      centrifuge: $(centrifuge --version | sed -n 1p | sed 's/^.*centrifuge-class version //')
  END_VERSIONS

Command exit status:
  17

Command output:
  (empty)

Command error:
  INFO:    Environment variable SINGULARITYENV_TMPDIR is set, but APPTAINERENV_TMPDIR is preferred
  INFO:    Environment variable SINGULARITYENV_NXF_TASK_WORKDIR is set, but APPTAINERENV_NXF_TASK_WORKDIR is preferred
  INFO:    Environment variable SINGULARITYENV_NXF_DEBUG is set, but APPTAINERENV_NXF_DEBUG is preferred
  WARNING: Skipping mount /var/apptainer/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
  (ERR): mkfifo(/tmp/46.inpipe1) failed.
  Exiting now ...

Work dir:
  /mnt/lustre/groups/nahnsen/nahkr648/test/work/56/78d3c43078646a260420d4b2e1a5b6

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
```

The `.command.log` and `.command.err` contain
```
INFO:    Environment variable SINGULARITYENV_TMPDIR is set, but APPTAINERENV_TMPDIR is preferred
INFO:    Environment variable SINGULARITYENV_NXF_TASK_WORKDIR is set, but APPTAINERENV_NXF_TASK_WORKDIR is preferred
INFO:    Environment variable SINGULARITYENV_NXF_DEBUG is set, but APPTAINERENV_NXF_DEBUG is preferred
WARNING: Skipping mount /var/apptainer/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
(ERR): mkfifo(/tmp/46.inpipe1) failed.
Exiting now ...
```

I guess it is related to the Centrifuge problem with the tmp dir as described in https://github.com/nf-core/taxprofiler/issues/422 (and solved in https://github.com/nf-core/taxprofiler/pull/444). 
Just wanted to mention it here, in case it is still relevant. 

### Command used and terminal output

```console
nextflow run nf-core/mag -r 2.5.4 -profile test -c myM3C.config --outdir results -resume
```
```


### Relevant files

_No response_

### System information

- Nextflow version 23.10.1
- HPC
- Slurm
- Singularity
- nf-core/mag 2.5.4

### PR DESCRIPTION
**Pull request by jfy133**
Closes #597 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
TODO: 

- CLean up output in results directory to match in docs
- Remove dumps
- Check CI fails above

But otherwise look slike output is expected now

File: nextflow_schema.json [hunk 1]
```diff
@@ -491,11 +491,15 @@
             "properties": {
                 "centrifuge_db": {
                     "type": "string",
+                    "format": "file-path",
+                    "pattern": ".*.tar.gz",
```
Comment (prototaxites) [inline nextflow_schema.json:495]: Just checking - looks like supplying a directory as a centrifuge directory is supported, but would this kick back anything that's not a tar.gz?

File: conf/modules.config [hunk 2]
```diff
@@ -239,7 +248,16 @@ process {
         ]
     }
 
-    withName: KRONA {
+    withName: KREPORT2KRONA_CENTRIFUGE {
+        publishDir = [
+            // path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" },
```
Comment (prototaxites) [inline conf/modules.config:253]: Is this meant to be commented out?

File: nextflow_schema.json (hunk 1 above)
Comment (jfy133) [reply nextflow_schema.json:495]: Ah crap yes, you're correct

File: conf/modules.config (hunk 2 above)
Comment (jfy133) [reply conf/modules.config:253]: Oops forgot to uncommend after setting enabled: fasle!

Comment (jfy133) [inline conf/modules.config:253]:
```suggestion
            path: { "${params.outdir}/Taxonomy/${meta.classifier}/${meta.id}" },
```

File: nextflow_schema.json (hunk 1 above)
Comment (jfy133) [inline nextflow_schema.json:495]:
```suggestion
```

**Review by prototaxites — APPROVED** [review]
Couple small comments - otherwise LGTM! (but don't have the latitude to test it myself right now)

**Pull request merged**